components/lib/crash/src/main/java/mozilla/components/lib/crash/service/MozillaSocorroService.kt

```diff
@@ -20,54 +37,197 @@ typealias GeckoCrashReporter = CrashReporter
+            sendPart(os, boundary, "BuildID", BuildConfig.MOZ_APP_BUILDID)
+            sendPart(os, boundary, "Vendor", BuildConfig.MOZ_APP_VENDOR)
+            sendPart(os, boundary, "ReleaseChannel", BuildConfig.MOZ_UPDATE_CHANNEL)
+            sendPart(os, boundary, "StartupTime", TimeUnit.MILLISECONDS.toSeconds(startTime)
+                    .toString())
+            sendPart(os, boundary, "CrashTime", TimeUnit.MILLISECONDS.toSeconds(
+                    System.currentTimeMillis()).toString())
+
+            val packageManager = applicationContext.packageManager
+            try {
+                val packageInfo = packageManager.getPackageInfo(applicationContext.packageName, 0)
+                sendPart(os, boundary, "InstallTime", TimeUnit.MILLISECONDS.toSeconds(
+                        packageInfo.lastUpdateTime).toString())
+            } catch (e: PackageManager.NameNotFoundException) {
+                Logger.error("Error getting package info", e)
+            }
+
+            throwable?.let {
+                sendPart(os, boundary, "JavaStackTrace", getExceptionStackTrace(it))
+            }
+
+            miniDumpFilePath?.let {
+                val minidumpFile = File(it)
+                sendFile(os, boundary, "upload_file_minidump", minidumpFile)
+                minidumpFile.delete()
+            }
+
+            val pid = android.os.Process.myPid()
+            val manager = applicationContext.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
+            manager.runningAppProcesses.filter { it.pid == pid }.forEach {
+                sendPart(os, boundary, "Android_ProcessName", it.processName)
+            }
+
+            sendPart(os, boundary, "Android_PackageName", applicationContext.packageName)
+            sendPart(os, boundary, "Android_Manufacturer", Build.MANUFACTURER)
+            sendPart(os, boundary, "Android_Model", Build.MODEL)
+            sendPart(os, boundary, "Android_Board", Build.BOARD)
+            sendPart(os, boundary, "Android_Brand", Build.BRAND)
+            sendPart(os, boundary, "Android_Device", Build.DEVICE)
+            sendPart(os, boundary, "Android_Display", Build.DISPLAY)
+            sendPart(os, boundary, "Android_Fingerprint", Build.FINGERPRINT)
+            sendPart(os, boundary, "Android_Hardware", Build.HARDWARE)
+            sendPart(os, boundary, "Android_Version", "${Build.VERSION.SDK_INT} (${Build.VERSION.CODENAME})")
+            os.write(("\r\n--$boundary--\r\n").toByteArray())
+            os.flush()
+            os.close()
+
+            BufferedReader(InputStreamReader(inputStream)).use {
+                val response = StringBuffer()
+
+                var inputLine = it.readLine()
+                while (inputLine != null) {
+                    response.append(inputLine)
+                    inputLine = it.readLine()
+                }
+                it.close()
+                Logger.info("Crash reported to Socorro: $response")
+            }
+        }
```

Comment:
This part looks like it can throw `IOException`?

Reply:
I'll check.  Thanks,